src/components/popover/bl-popover.ts

```diff
@@ -0,0 +1,200 @@
+
+    this.popoverAutoUpdateCleanup && this.popoverAutoUpdateCleanup();
+  }
+  private _middleware: Middleware[] = [];
+
+  protected firstUpdated() {
+    this._middleware.push(
+      offset(this.offset),
+      flip(),
+      inline(),
+      arrow({ element: this.arrow, padding: 5 })
+    );
+    if (this.shift) this._middleware.push(shift({ padding: 5 }));
+
+    if (this.fitSize) {
+      this._middleware.push(
+        size({
+          apply(args: MiddlewareState) {
+            if(args.elements.floating && args.elements.reference){
+              Object.assign(args.elements.floating.style, {
+                width: `${args.elements.reference.getBoundingClientRect().width}px`,
+              });
+            }
+          },
+        })
+      );
+    }
+  }
+
+  private popoverAutoUpdateCleanup: () => void;
+
+  private setPopover() {
+    if(this.target){
+      this.popoverAutoUpdateCleanup = autoUpdate(this.target as Element, this.popover, () => {
+        computePosition(this.target as Element, this.popover, {
+          placement: this.placement,
+          strategy: 'fixed',
+          middleware: this._middleware,
+        }).then(({ x, y, placement, middlewareData }) => {
+          Object.assign(this.popover.style, {
+            left: `${x}px`,
+            top: `${y}px`,
+          });
+
+          if (middlewareData.arrow) {
+            const { x: arrowX, y: arrowY } = middlewareData.arrow;
+
+            Object.assign(this.arrow.style, {
+              left: `${arrowX}px`,
+              top:`${arrowY}px`,
+            });
+
+            const arrowFlipDirections = {
+              top: 'bottom',
+              right: 'left',
+              bottom: 'top',
+              left: 'right',
+            };
+            const popoverPlacement = placement.split('-')[0] as keyof typeof arrowFlipDirections;
+            const arrowDirection = arrowFlipDirections[popoverPlacement];
+
+            this.arrow.style.setProperty(arrowDirection, '-4px');
+          }
+        });
+      });
+    }
+  }
+
+  @property()
+  get target(): string | Element {
+    return this._target;
+  }
+
+  set target(value: string | Element) {
+    if (typeof value === 'string') {
+      this._target = document.getElementById(value) as Element;
+    } else {
+      this._target = value;
+    }
+  }
+
+  /**
+   * Shows popover
+   */
+  show() {
+    this._visible = true;
+    this.setPopover();
+    this.onBlPopoverShow('Show event fired!');
+  }
+
+  /**
+   * Hides popover
+   */
+  hide() {
+    this._visible = false;
+    this.onBlPopoverHide('Hide event fired!');
+  }
+
+  /**
+   * Gives the visibility status of the popover
+   */
+  get visible(): boolean {
+    return this._visible;
+  }
+
+  render(): TemplateResult {
+    const classes = classMap({
+      popover: true,
+      visible: this._visible,
+    });
+
+    return html`<div class=${classes}>
+      <slot id="popover" role="popover" aria-live=${this._visible ? 'polite' : 'off'}></slot>
```

Comment:
Apparently [there is no ARIA role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles) named `popover`. I didn't make a detailed investigation about proper A11Y guidance over popover elements, but we need to check the best practices here.

Reply:
So, should I remove role attribute temporarily?

Reply:
Yes, I think so. But better would be to find a proper solution.